terraform/api_gateway.tf

```diff
@@ -0,0 +1,54 @@
+resource "aws_api_gateway_rest_api" "redirector_api" {
+  name        = "${var.redirector_lambda_name}"
+  description = "API Gateway for the redirector lambda function."
+}
+
+resource "aws_api_gateway_method" "proxy_root" {
+  rest_api_id   = "${aws_api_gateway_rest_api.redirector_api.id}"
+  resource_id   = "${aws_api_gateway_rest_api.redirector_api.root_resource_id}"
+  http_method   = "GET"
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_integration" "lambda_root" {
+  rest_api_id = "${aws_api_gateway_rest_api.redirector_api.id}"
+  resource_id = "${aws_api_gateway_method.proxy_root.resource_id}"
+  http_method = "${aws_api_gateway_method.proxy_root.http_method}"
+
+  integration_http_method = "POST"
+  type                    = "AWS_PROXY"
+  uri                     = "${aws_lambda_function.redirector.invoke_arn}"
+}
+
+resource "aws_api_gateway_deployment" "redirector_deployment" {
+  depends_on = [
+    "aws_api_gateway_integration.lambda_root",
+  ]
+  rest_api_id = "${aws_api_gateway_rest_api.redirector_api.id}"
+  stage_name  = "${var.api_gateway_stage_name}"
+}
+
+resource "aws_lambda_permission" "lambda_api" {
+  depends_on = [
+    "aws_lambda_function.redirector",
+    "aws_api_gateway_rest_api.redirector_api",
+    "aws_api_gateway_method.proxy_root"
+  ]
```

Comment:
I'm surprised that we need these `depends_on` thingies. Usually terraform is smart enough to figure it out automatically 🤔